### PR TITLE
ISPN-15262 Allow blocking calls on distributed indexer lock

### PR DESCRIPTION
**Pull request by fax4ever**
https://issues.redhat.com/browse/ISPN-15262

### CODE REVIEW

File: query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java [hunk 1]
```diff
@@ -12,5 +13,6 @@ public class QueryBlockHoundIntegration implements BlockHoundIntegration {
    public void applyTo(BlockHound.Builder builder) {
       // Loading a service may require opening a file from classpath
       builder.allowBlockingCallsInside(NamedSPILoader.class.getName(), "reload");
+      builder.allowBlockingCallsInside(DistributedIndexerLock.class.getName(), "lock");
```
Comment (wburns) [inline query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java:16]: Is this really for `DistributedIndexerLock#getLock`?

**Review by wburns — CHANGES_REQUESTED** [review]
We should also get a JIRA with the description of what tests this fixes.

File: query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java (hunk 1 above)
Comment (fax4ever) [reply query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java:16]: actually is for `ClusteredLock#tryLock`

**Comment by fax4ever** [comment]
@wburns created

File: query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java (hunk 1 above)
Comment (fax4ever) [reply query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java:16]: Actually it seems that both the methods (getLock and tryLock) can lead to the execution of `java.util.concurrent.locks.ReentrantLock#lock` for instance. I'm sending the two stack traces.

Comment (fax4ever) [reply query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java:16]: At this point I would add the entire `DistributedIndexerLock#lock` that is supposed to cover both. WDYT?

Comment (fax4ever) [reply query/src/main/java/org/infinispan/query/internal/QueryBlockHoundIntegration.java:16]: Ah! it seems what I did it.... :P

**Comment by fax4ever** [comment]
Closing this PR. This is the root issue => https://github.com/infinispan/infinispan/pull/11433

**Comment by fax4ever** [comment]
https://github.com/infinispan/infinispan/pull/11433

**Pull request closed without merging**